src/zerocoin_v3.cpp

```diff
@@ -595,21 +595,7 @@ bool ConnectBlockSigma(
     return true;
 }
 
-bool GetOutPoint(COutPoint& outPoint, sigma::PublicCoin pubCoin) {
-
-    sigma::CSigmaState *sigmaState = sigma::CSigmaState::GetState();
-    auto mintedCoinHeightAndId = sigmaState->GetMintedCoinHeightAndId(pubCoin);
-    int mintHeight = mintedCoinHeightAndId.first;
-    int coinId = mintedCoinHeightAndId.second;
-    if(mintHeight==-1 && coinId==-1)
-        return false;
-
-    // get block containing mint
-    CBlockIndex *mintBlock = chainActive[mintHeight];
-    CBlock block;
-    if(!ReadBlockFromDisk(block, mintBlock, ::Params().GetConsensus()))
-        LogPrintf("can't read block from disk.\n");
-
+bool GetOutPointFromBlock(COutPoint& outPoint, GroupElement pubCoinValue, CBlock block){
```

Comment:
`CBlock` is too large to pass by value. This one must be passed by reference, so please change it to `const CBlock&`. Also will be better to fix this for `pubCoinValue` too.